src/sync/sync_permission.hpp

```diff
@@ -93,12 +93,24 @@ class PermissionResults {
         return m_results.async(std::move(target));
     }
 
-    // Create a new instance by further filtering or sorting this instance.
+    // Create a new instance by further filtering this instance.
     PermissionResults filter(Query&& q) const
     {
         return PermissionResults(m_results.filter(std::move(q)));
     }
 
+    // Create a new instance by sorting this instance.
+    PermissionResults sort(SortDescriptor&& s) const
+    {
+        return PermissionResults(m_results.sort(std::move(s)));
+    }
+
+    // Get the results.
+    Results& results()
```

Comment:
@tgoyne performance isn't really an issue here and even if it were the difference seems inconsequential. Also seems like it is about the same amount of code, ie ~1 line. In any case I'm more concerned with the fact that we are accessing something which is meant to be private/hidden behind an abstraction.